phpBB/phpbb/db/tools/tools_interface.php

```diff
@@ -22,111 +22,114 @@ interface tools_interface
+	 *
+	 * @return array        Array of column names (all lower case)
 	 */
-	public function sql_list_columns($table_name);
+	public function sql_list_columns(string $table_name): array;
 
 	/**
 	 * Check whether a specified column exist in a table
 	 *
-	 * @param string	$table_name		Table to check
-	 * @param string	$column_name	Column to check
-	 * @return bool		True if column exists, false otherwise
+	 * @param string $table_name  Table to check
+	 * @param string $column_name Column to check
+	 *
+	 * @return bool        True if column exists, false otherwise
 	 */
-	public function sql_column_exists($table_name, $column_name);
+	public function sql_column_exists(string $table_name, string $column_name): bool;
 
 	/**
 	 * Add new column
 	 *
-	 * @param string	$table_name		Table to modify
-	 * @param string	$column_name	Name of the column to add
-	 * @param array		$column_data	Column data
-	 * @param bool		$inline			Whether the query should actually be run,
-	 *						or return a string for adding the column
-	 * @return array|true	Statements to run, or true if the statements have been executed
+	 * @param string $table_name  Table to modify
+	 * @param string $column_name Name of the column to add
+	 * @param array  $column_data Column data
+	 *
+	 * @return bool|string[]    True if the statements have been executed
 	 */
-	public function sql_column_add($table_name, $column_name, $column_data, $inline = false);
+	public function sql_column_add(string $table_name, string $column_name, array $column_data);
 
 	/**
 	 * Change column type (not name!)
 	 *
-	 * @param string	$table_name		Table to modify
-	 * @param string	$column_name	Name of the column to modify
-	 * @param array		$column_data	Column data
-	 * @param bool		$inline			Whether the query should actually be run,
-	 *						or return a string for modifying the column
-	 * @return array|true	Statements to run, or true if the statements have been executed
+	 * @param string $table_name  Table to modify
+	 * @param string $column_name Name of the column to modify
+	 * @param array  $column_data Column data
+	 *
+	 * @return bool|string[]    True if the statements have been executed
 	 */
-	public function sql_column_change($table_name, $column_name, $column_data, $inline = false);
+	public function sql_column_change(string $table_name, string $column_name, array $column_data);
 
 	/**
 	 * Drop column
 	 *
-	 * @param string	$table_name		Table to modify
-	 * @param string	$column_name	Name of the column to drop
-	 * @param bool		$inline			Whether the query should actually be run,
-	 *						or return a string for deleting the column
-	 * @return array|true	Statements to run, or true if the statements have been executed
+	 * @param string $table_name  Table to modify
+	 * @param string $column_name Name of the column to drop
+	 *
+	 * @return bool|string[]    True if the statements have been executed
 	 */
-	public function sql_column_remove($table_name, $column_name, $inline = false);
```

Comment:
Not sure if we want to keep `$inline`, I guess it worth to think about it.

Reply:
we don't need it anymore. It was used by perform_schema_change